Templates/Equality.stencil

```diff
@@ -0,0 +1,20 @@
+import Foundation
+
+{% for type in types.implementing.AutoEquatable|class|!annotated:"skipEquality" %}
+extension {{ type.name }} {
+    /// :nodoc:
+    @objc override {{ type.accessLevel }} func isEqual(to object: Any?) -> Bool {
+      guard let rhs = object as? {{ type.name }} else { return false }
+      {% for variable in type.storedVariables %}
+      {% if variable|annotated:"notEquatableDictionary" %}
+      if !NSDictionary(dictionary: self.{{ variable.name}} {% if variable.typeName.isOptional %}?? [:]{% endif %}).isEqual(to: rhs.{{ variable.name }} {% if variable.typeName.isOptional %}?? [:]{% endif %}) { return false }
```

Comment:
I would just check for `[String: Any]` type, that's the only type you need this for. Just use `actualTypeName` instead of `typeName` to include typealias too.

Reply:
How can I write that condition @ilyapuchka? I tried this but doesn't seem to work:

```
{% if variable|actualTypeName:"[String: Any]" %}
```